File: p010/p10.php
<?php

/**
The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.
Find the sum of all the primes below two million.
**/

require_once "../lib/prime.php";

$limit = 2000000;
$sum = 2;
for($i = 3; $i <= $limit; $i += 2 ) {
	if(is_prime($i)) {
		$sum += $i;
	}
}

print $sum . "\n";

?>

File: p010/p10.c
#include <stdio.h>
#include <math.h>

#include "../lib/prime.h"

#define LIMIT 2000000

typedef unsigned long long i64;

int main(int argc, char* argv[]) {
	
	sieve_t sieve = prime_sieve(LIMIT);

	i64 total = 0;
	for(int i = 0; i < sieve.num; i++) {
			total += sieve.primes[i];
	}
	
	printf("%llu", total);

	return 0;
}

